File: app/src/main/java/com/meta4projects/fldfloatingdictionary/room/database/BookmarkDatabase.java
package com.meta4projects.fldfloatingdictionary.room.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.meta4projects.fldfloatingdictionary.room.daos.BookMarkDao;
import com.meta4projects.fldfloatingdictionary.room.entities.BookMark;

@Database(entities = {BookMark.class}, version = 1, exportSchema = false)
public abstract class BookmarkDatabase extends RoomDatabase {

    private static volatile BookmarkDatabase INSTANCE;

    public static BookmarkDatabase getINSTANCE(final Context context) {
        if (INSTANCE == null) {
            synchronized (BookmarkDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = getDataBase(context);
                }
            }
        }

        return INSTANCE;
    }

    private static BookmarkDatabase getDataBase(final Context context) {
        return Room.databaseBuilder(context.getApplicationContext(),
                BookmarkDatabase.class, "bookmark_database").build();
    }

    public abstract BookMarkDao bookMarkDao();
}

File: app/src/main/java/com/meta4projects/fldfloatingdictionary/room/database/BookmarkDatabase.kt
package com.meta4projects.fldfloatingdictionary.room.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import com.meta4projects.fldfloatingdictionary.room.daos.BookMarkDao
import com.meta4projects.fldfloatingdictionary.room.entities.BookMark

@Database(entities = [BookMark::class], version = 2, exportSchema = false)
abstract class BookmarkDatabase : RoomDatabase() {
    abstract fun bookMarkDao(): BookMarkDao

    companion object {
        @Volatile
        private var INSTANCE: BookmarkDatabase? = null

        @JvmStatic
        fun getINSTANCE(context: Context): BookmarkDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = getDataBase(context)
                INSTANCE = instance
                instance
            }
        }

        private fun getDataBase(context: Context): BookmarkDatabase {
            return databaseBuilder(context.applicationContext, BookmarkDatabase::class.java, "bookmark_database").fallbackToDestructiveMigration().build()
        }
    }
}